### PR TITLE
Use default values instead of holes for strings and numbers

### PR DESCRIPTION
**Pull request by shmish111**
This will make it possible to have holes in Blockly since its not possible to have null values (hence a hole) for strings and numbers.

Note that for linting to work correctly, Marlowe.Holes.Timeout records its position.

### CODE REVIEW

File: marlowe-playground-client/src/Marlowe/Holes.purs [hunk 1]
```diff
@@ -172,17 +174,17 @@ getMarloweConstructors ContractType =
...
+getMarloweConstructors BoundType = Map.singleton "Bound" [ DefaultNumber zero, DefaultNumber zero ]
 
-getMarloweConstructors TokenType = Map.singleton "Token" [ DataArg "currency", DataArg "token" ]
+getMarloweConstructors TokenType = Map.singleton "Token" [ DefaultString "", DefaultString "" ]
 
 getMarloweConstructors PartyType =
   Map.fromFoldable
-    [ (Tuple "PK" [ DataArg "pubKey" ])
+    [ (Tuple "PK" [ DefaultString "pubKey" ])
     , (Tuple "Role" [ DataArg "token" ])
```
Comment (palas) [inline marlowe-playground-client/src/Marlowe/Holes.purs:188]: This `Role` expansion is expanding to something that doesn't parse. Not sure were in the code the problem is

File: marlowe-playground-client/src/Marlowe/Holes.purs [hunk 2]
```diff
@@ -117,9 +119,9 @@ getMarloweConstructors BigIntegerType = Map.singleton "Integer" [ NewtypeArg ]
 
 getMarloweConstructors SlotType = Map.singleton "Integer" [ NewtypeArg ]
 
-getMarloweConstructors AccountIdType = Map.singleton "AccountId" [ DataArg "accNumber", DataArg "accHolder" ]
+getMarloweConstructors AccountIdType = Map.singleton "AccountId" [ DefaultNumber zero, DataArg "accHolder" ]
 
-getMarloweConstructors ChoiceIdType = Map.singleton "ChoiceId" [ DataArg "choiceNumber", DataArg "choiceOwner" ]
+getMarloweConstructors ChoiceIdType = Map.singleton "ChoiceId" [ DefaultString "choiceNumber", DataArg "choiceOwner" ]
 
 getMarloweConstructors ValueIdType = Map.singleton "ValueId" [ NewtypeArg ]
```
Comment (palas) [inline marlowe-playground-client/src/Marlowe/Holes.purs:126]: I think `ValueId` has the `OverloadedString` too. It seems hole expansion already fails in alpha too, not sure

File: marlowe-playground-client/src/Marlowe/Holes.purs (hunk 1 above)
Comment (shmish111) [reply marlowe-playground-client/src/Marlowe/Holes.purs]: I've now added tests that ensure that all `MarloweType` values are tested.

File: marlowe-playground-client/src/Marlowe/Holes.purs (hunk 2 above)
Comment (shmish111) [reply marlowe-playground-client/src/Marlowe/Holes.purs]: The new tests cover this (plus I've fixed it)

**Pull request merged**